### PR TITLE
fix: form item Input size error with prefix or suffix

### LINKED ISSUE #21288: Input: form item Input size error with prefix and suffix
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/withered-mountain-tjnhj)

### Steps to reproduce
switch the form size,

### What is expected?
Input size works fine with prefix and suffix

### What is actually happening?
not works

| Environment | Info |
|---|---|
| antd | 4.0.0-rc.3 |
| React | latest |
| System | MacOS latest |
| Browser | Chrome latest |

### PR DESCRIPTION
**Pull request by yoyo837**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix #21288

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |    修复带图标的Input 和 Input.Password  在Form 下不支持通过SizeContext.Consumer 响应正确的尺寸的问题     |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/input/demo/size.md](https://github.com/ant-design/ant-design/blob/fix-input-size-with-prefix-and-suffix/components/input/demo/size.md)

### CODE REVIEW

**Comment by yoyo837** [comment]
Any plans for the rc.4 release? 🤔

File: components/input/style/affix.less [hunk 1]
```diff
@@ -7,6 +7,7 @@
   &-affix-wrapper {
     .input();
     display: inline-flex;
+    width: auto;
```
Comment (afc163) [inline components/input/style/affix.less]: 为什么要加这个。

Comment (yoyo837) [reply components/input/style/affix.less]: image

因为一旦加了prefix or suffix 就自动 100%了 是不是不太合理？

Comment (yoyo837) [reply components/input/style/affix.less]: 恢复吗？

Comment (yoyo837) [reply components/input/style/affix.less]: Select 在 form 下宽度是 `.ant-form-item .ant-select, .ant-form-item .ant-cascader-picker` 的样式 加宽的，感觉互相有点不统一。

Comment (afc163) [reply components/input/style/affix.less]: 看下 3.x 的逻辑是什么，保持一致就行。

Comment (yoyo837) [reply components/input/style/affix.less]: 3.x 一致。我撤销它！

**Pull request merged**